### PR TITLE
feat: add isEntryNode property to ActionBase

### PR DESCRIPTION
**Pull request by tpmsr**
Adds isEntryNode to actions.  The intent for this field is to instruct the backend to generate out-of-domain utterances that resolve to this action node.

This is relevant for bots that have an initial disambiguation node such as "How can I help with your printer today ?".

### CODE REVIEW

File: src/Action.ts [hunk 1]
```diff
@@ -79,41 +83,52 @@ export class ActionBase {
...
-    } else if (ActionTypes.API_LOCAL === action.actionType) {
-      let actionPayload = JSON.parse(action.payload) as ActionPayload
-      return actionPayload.payload
-    } else if (ActionTypes.DISPATCH === action.actionType) {
-      // TODO: Another reason to schema refactor...
-      let actionPayload = JSON.parse(action.payload) as DispatchPayload
-      return `${ActionTypes.DISPATCH}: ${actionPayload.modelName}`
+    switch (action.actionType) {
+      case ActionTypes.TEXT:
+        {
```
Comment (msft-shahins) [inline src/Action.ts]: nit: there is no official typescript style guideline and not sure what tslint enforces here but prefer to have `{` in the same line as case statement to make it similar to usage of `{` in other constructs. 

```javascript 
 case ActionTypes.TEXT: {
       /**
       * For backwards compatibility check if payload is of new TextPayload type	   
```

Comment (tpmsr) [reply src/Action.ts]: Done

File: src/Action.ts [hunk 2]
```diff
@@ -38,6 +38,9 @@ export class ActionBase {
   createdDateTime: string
   payload: string
   isTerminal: boolean
+  // If true, CL will generate out of domain example utterances that point to this action.
+  // There should be at most one action in a model with this flag set to `true`.
```
Comment (LarsLiden) [inline src/Action.ts]: Will you be adding a check for this in the UI code?

File: src/Action.ts [hunk 3]
```diff
@@ -79,41 +83,52 @@ export class ActionBase {
...
+          } catch (e) {
+            const error = e as Error
+            throw new Error(
+              `Error when attempting to parse text action payload. This might be an old action which was saved as a string.  Please create a new action. ${
+              error.message
+              }`
+            )
+          }
+        }
+      case ActionTypes.END_SESSION:
```
Comment (LarsLiden) [inline src/Action.ts]: Note that the brackets are actually optional for switch statements in Typescript.  Style-wise we've been choosing not to use them

Comment (tpmsr) [reply src/Action.ts]: IIRC there was some warning about variable redeclaration without them here.  Case statements apparently don't create block scope in JS / TS.

File: src/Action.ts (hunk 2 above)
Comment (tpmsr) [reply src/Action.ts]: Can do!

Comment (msft-shahins) [reply src/Action.ts]: thought that we are not exposing this in UI and this is only used for .dialog import.

File: src/Action.ts (hunk 3 above)
Comment (msft-shahins) [reply src/Action.ts]: ES6 doesn't create separate scopes for each case statements unless you make them different scope by using braces: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch#Block-scope_variables_within_switch_statements

Comment (LarsLiden) [reply src/Action.ts]: Yeah, you can't re-use var names if you don't use brackets.  I usually opt for using diff var names

File: src/Action.ts (hunk 2 above)
Comment (LarsLiden) [reply src/Action.ts]: In short term yes.  However weird to have validity check only mentioned in a comment.  There should at least be another comment in the action creator that this needs to be checked

**Pull request merged**